src/sinol_make/commands/gen/gen_util.py

```diff
@@ -0,0 +1,124 @@
+import glob
+import os
+
+import argparse
+import signal
+import subprocess
+import sys
+
+from sinol_make import util
+from sinol_make.helpers import compiler, package_util, compile
+
+
+def get_ingen(task_id=None, ingen_path=None):
+    """
+    Find ingen source file in `prog/` directory.
+    If `ingen_path` is specified, then it will be used (if exists).
+    :param task_id: task id, for example abc. If None, then
+                    will return any ingen matching "*ingen.*"
+    :param ingen_path: path to ingen source file
+    :return: path to ingen source file or None if not found
+    """
+
+    if ingen_path is not None:
+        if os.path.exists(ingen_path):
+            return ingen_path
+        else:
+            return None
+
+    if task_id is None:
+        task_id = '*'
+    ingen = glob.glob(os.path.join(os.getcwd(), 'prog', task_id + 'ingen.*'))
+    if len(ingen) == 0:
+        return None
+
+    # Makefiles prioritize non .sh ingens.
+    if os.path.splitext(ingen[0])[1] == '.sh' and len(ingen) > 1:
+        return ingen[1]
+    else:
+        return ingen[0]
+
+
+def compile_ingen(ingen_path: str, args: argparse.Namespace, weak_compilation_flags=False):
+    """
+    Compiles ingen and returns path to compiled executable and path to compile log.
+    """
+    compilers = compiler.verify_compilers(args, [ingen_path])
+    return compile.compile_file(ingen_path, package_util.get_executable(ingen_path), compilers, weak_compilation_flags)
+
+
+def get_correct_solution(task_id):
+    """
+    Returns path to correct solution for given task.
+    :param task_id: task id, for example abc
+    :return: path to correct solution or None if not found
+    """
+    correct_solution = glob.glob(os.path.join(os.getcwd(), 'prog', task_id + '.*'))
+    if len(correct_solution) == 0:
+        return None
+    return correct_solution[0]
+
+
+def compile_correct_solution(solution_path: str, args: argparse.Namespace, weak_compilation_flags=False):
+    """
+    Compiles correct solution and returns path to compiled executable and path to compile log.
+    """
+    compilers = compiler.verify_compilers(args, [solution_path])
+    return compile.compile_file(solution_path, package_util.get_executable(solution_path), compilers,
+                                weak_compilation_flags)
+
+
+def run_ingen(ingen_exe):
+    """
+    Runs ingen and generates all input files.
+    :param ingen_exe: path to ingen executable
+    :return: True if ingen was successful, False otherwise
+    """
+    shell = False
+    if os.path.splitext(ingen_exe)[1] == '.sh':
+        shell = True
```

Comment:
```suggestion
    shell = os.path.splitext(ingen_exe)[1] == '.sh'
```
Also, `is_shell` might be a better name, as currently the name suggest that this variable contains the shell type, or something similar.